Makefile

```diff
@@ -61,9 +61,13 @@ fmt:
 	go fmt ./pkg/... ./cmd/...
 
 .PHONY: goimports
-    goimports: ## Go fmt your code
+goimports: ## Go fmt your code
 	hack/goimports.sh .
 
+.PHONY: generate
+generate: ## generate code (e.g. deepcopy)
+	./hack/update-generated-deep-copies.sh
```

Comment:
Just a nitpick, but why generating binary here? It could be done with something close to:
```bash
go run ./vendor/k8s.io/code-generator/cmd/deepcopy-gen
                --go-header-file "hack/boilerplate/boilerplate.go.txt" \
                --input-dirs "./pkg/apis/azureprovider/v1beta1" \
                --output-file-base zz_generated.deepcopy
```

Reply:
So the deepcopy generator can be run independent of the Makefile